syntax = "proto3";

package autotiler;

message AutoTilerMap {
    enum BaseTerrain {
        PLAINS = 0;
        GRASSLAND = 1;
        DESERT = 2;
        TUNDRA = 3;
        SNOW = 4;

        COAST = 5;
        OCEAN = 6;

        PLAINS_HILLS = 7;
        GRASSLAND_HILLS = 8;
        DESERT_HILLS = 9;
        TUNDRA_HILLS = 10;
        SNOW_HILLS = 11;
    }

    enum Improvement {
        NONE = 0;
        CITY = 1;
    }

    message Tile {
        int32 row = 1;
        int32 col = 2;

        BaseTerrain baseTerrain = 3;
        Improvement improvement = 4;

        int32 food = 10;
        int32 production = 11;
        int32 gold = 12;
        int32 science = 13;
        int32 culture = 14;
        int32 faith = 15;

        int32 yieldValue = 20;
    }

    int32 rows = 1;
    int32 cols = 2;
    repeated Tile tiles = 3;
}